### PR TITLE
Spellchecking compiler comments

### PR DESCRIPTION
**Pull request by nyurik**
This PR cleans up the rest of the spelling mistakes in the compiler comments. This PR does not change any literal or code spelling issues.

### CODE REVIEW

**Comment by compiler-errors** [comment]
Wondering if this should be split up into several PRs by crate (or sets of crates) so this is easier to land...

**Comment by nyurik** [comment]
> Wondering if this should be split up into several PRs by crate (or sets of crates) so this is easier to land...

@compiler-errors I could break it up a bit, but that's 28 crates - I think that's a bit excessive to have that many PRs with a few lines each?

File: compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs [hunk 1]
```diff
@@ -1650,7 +1650,7 @@ impl<'p, 'tcx> fmt::Debug for DeconstructedPat<'p, 'tcx> {
                     }
 
                     // Without `cx`, we can't know which field corresponds to which, so we can't
-                    // get the names of the fields. Instead we just display everything as a suple
+                    // get the names of the fields. Instead we just display everything as a simple
```
Comment (compiler-errors) [inline compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs:1653]: tuple

File: compiler/rustc_infer/src/infer/region_constraints/leak_check.rs [hunk 2]
```diff
@@ -243,7 +243,7 @@ impl<'me, 'tcx> LeakCheck<'me, 'tcx> {
             // Update minimum universe of scc1.
             self.scc_universes[scc1] = scc1_universe;
 
-            // At this point, `scc_placholder[scc1]` stores the placeholder that
+            // At this point, `scc_placeholder[scc1]` stores the placeholder that
```
Comment (compiler-errors) [inline compiler/rustc_infer/src/infer/region_constraints/leak_check.rs:246]: scc_placeholders

File: compiler/rustc_parse/src/parser/expr.rs [hunk 3]
```diff
@@ -276,7 +276,7 @@ impl<'a> Parser<'a> {
                 lhs = self.parse_assoc_op_ascribe(lhs, lhs_span)?;
                 continue;
             } else if op == AssocOp::DotDot || op == AssocOp::DotDotEq {
-                // If we didn’t have to handle `x..`/`x..=`, it would be pretty easy to
+                // If we did not have to handle `x..`/`x..=`, it would be pretty easy to
```
Comment (compiler-errors) [inline compiler/rustc_parse/src/parser/expr.rs:279]: this one seems kinda arbitrary, but ok

File: compiler/rustc_passes/src/stability.rs [hunk 4]
```diff
@@ -144,7 +144,7 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
         }
 
         if !self.tcx.features().staged_api {
-            // Propagate unstability.  This can happen even for non-staged-api crates in case
+            // Propagate instability.  This can happen even for non-staged-api crates in case
```
Comment (compiler-errors) [inline compiler/rustc_passes/src/stability.rs:147]: I think this should stay "unstability" -- it's technically not a word, but the attribute is called `unstable`.

**Comment by compiler-errors** [comment]
Looks fine modulo comments I left above. Can you address those? Otherwise I think it's fine to land -- I see that @lcnr had approved some similar typo PRs of yours recently.

File: compiler/rustc_apfloat/src/ieee.rs [hunk 5]
```diff
@@ -2646,7 +2646,7 @@ mod sig {
 
         // Convert the result having "2 * precision" significant-bits back to the one
         // having "precision" significant-bits. First, move the radix point from
-        // poision "2*precision - 1" to "precision - 1". The exponent need to be
+        // position "2*precision - 1" to "precision - 1". The exponent need to be
```
Comment (bjorn3) [inline compiler/rustc_apfloat/src/ieee.rs:2649]: This should probably not be changed. The license situation around rustc_apfloat is unclear.

File: compiler/rustc_codegen_cranelift/src/pretty_clif.rs [hunk 6]
```diff
@@ -23,7 +23,7 @@
 //!     ss2 = explicit_slot 8 ; _4: (&&[u16],) size=8 align=8,8
 //!     sig0 = (i64, i64, i64) system_v
 //!     sig1 = (i64, i64, i64) system_v
-//!     fn0 = colocated u0:6 sig1 ; Instance { def: Item(DefId(0/0:31 ~ example[8787]::{{impl}}[1]::call_mut[0])), substs: [ReErased, ReErased] }
+//!     fn0 = collocated u0:6 sig1 ; Instance { def: Item(DefId(0/0:31 ~ example[8787]::{{impl}}[1]::call_mut[0])), substs: [ReErased, ReErased] }
```
Comment (bjorn3) [inline compiler/rustc_codegen_cranelift/src/pretty_clif.rs:26]: No, this means co-located. Cranelift ir spells it as colocated.

File: compiler/rustc_mir_build/src/build/matches/mod.rs [hunk 7]
```diff
@@ -444,7 +444,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             // we lower the guard.
             let target_block = self.cfg.start_new_block();
             let mut schedule_drops = true;
-            // We keep a stack of all of the bindings and type asciptions
+            // We keep a stack of all of the bindings and type descriptions
             // from the parent candidates that we visit, that also need to
```
Comment (bjorn3) [inline compiler/rustc_mir_build/src/build/matches/mod.rs:448]: The previous comment was correct. A type ascription is the `: ty` in `expr: ty` or `let pat: ty`.

File: compiler/rustc_mir_build/src/build/matches/test.rs [hunk 8]
```diff
@@ -234,7 +234,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     };
                     TerminatorKind::if_(self.tcx, Operand::Copy(place), true_bb, false_bb)
                 } else {
-                    // The switch may be inexhaustive so we have a catch all block
+                    // The switch may be inexhaustible so we have a catch all block
```
Comment (bjorn3) [inline compiler/rustc_mir_build/src/build/matches/test.rs:237]: We use exhaustive as name inside code elsewhere.

File: compiler/rustc_mir_build/src/build/matches/mod.rs (hunk 7 above)
Comment (compiler-errors) [reply compiler/rustc_mir_build/src/build/matches/mod.rs:448]: `s/asciptions/ascriptions` not descriptions

File: compiler/rustc_parse/src/parser/expr.rs (hunk 3 above)
Comment (nyurik) [reply compiler/rustc_parse/src/parser/expr.rs:279]: i just realized this was a backtick, that's why spellcheck got confused. I'll change it back to a regular tick.

File: compiler/rustc_infer/src/infer/outlives/components.rs [hunk 9]
```diff
@@ -125,7 +125,7 @@ fn compute_components<'tcx>(
             // regionck more ways to prove that it holds. However,
             // regionck is not (at least currently) prepared to deal with
             // higher-ranked regions that may appear in the
-            // trait-ref. Therefore, if we see any higher-ranke regions,
+            // trait-ref. Therefore, if we see any higher-rank regions,
```
Comment (bjorn3) [inline compiler/rustc_infer/src/infer/outlives/components.rs:128]: *ranked

File: compiler/rustc_middle/src/mir/interpret/error.rs [hunk 10]
```diff
@@ -234,7 +234,7 @@ pub enum UndefinedBehaviorInfo<'tcx> {
     },
     /// Something was divided by 0 (x / 0).
     DivisionByZero,
-    /// Something was "remainded" by 0 (x % 0).
+    /// Something was "remaineded" by 0 (x % 0).
```
Comment (bjorn3) [inline compiler/rustc_middle/src/mir/interpret/error.rs:237]: I think the old version was correct. Not sure though.

File: compiler/rustc_middle/src/ty/sty.rs [hunk 11]
```diff
@@ -284,7 +284,7 @@ static_assert_size!(TyKind<'_>, 32);
 /// - 'l0...'li and T0...Tj are the generic parameters
 ///   in scope on the function that defined the closure,
 /// - CK represents the *closure kind* (Fn vs FnMut vs FnOnce). This
-///   is rather hackily encoded via a scalar type. See
+///   is rather hacky encoded via a scalar type. See
```
Comment (bjorn3) [inline compiler/rustc_middle/src/ty/sty.rs:287]: I think the old version was correcter. At least the new version doesn't cover the intended meaning IMO.

File: compiler/rustc_mir_transform/src/simplify_try.rs [hunk 12]
```diff
@@ -72,7 +72,7 @@ struct ArmIdentityInfo<'tcx> {
     /// (StorageLive index,, StorageDead index, Local)
     storage_stmts: Vec<(usize, usize, Local)>,
 
-    /// The statements that should be removed (turned into nops)
+    /// The statements that should be removed (turned into noops)
```
Comment (bjorn3) [inline compiler/rustc_mir_transform/src/simplify_try.rs:75]: The StatementKind variant is called Nop.

File: compiler/rustc_target/src/spec/mod.rs [hunk 13]
```diff
@@ -1255,7 +1255,7 @@ pub struct TargetOptions {
     /// handling COFF object files with more than 2<sup>15</sup> sections. Since each weak
     /// symbol needs its own COMDAT section, weak linkage implies a large
     /// number sections that easily exceeds the given limit for larger
-    /// codebases. Consequently we want a way to disallow weak linkage on some
+    /// codebase. Consequently we want a way to disallow weak linkage on some
```
Comment (bjorn3) [inline compiler/rustc_target/src/spec/mod.rs:1258]: This should remain plural.

File: compiler/rustc_trait_selection/src/traits/codegen.rs [hunk 14]
```diff
@@ -46,8 +46,8 @@ pub fn codegen_fulfill_obligation<'tcx>(
             Ok(Some(selection)) => selection,
             Ok(None) => {
                 // Ambiguity can happen when monomorphizing during trans
-                // expands to some humongo type that never occurred
-                // statically -- this humongo type can then overflow,
+                // expands to some humongous type that never occurred
+                // statically -- this humongous type can then overflow,
```
Comment (bjorn3) [inline compiler/rustc_trait_selection/src/traits/codegen.rs:50]: Not sure about these.

File: compiler/rustc_mir_build/src/build/matches/test.rs (hunk 8 above)
Comment (RalfJung) [reply compiler/rustc_mir_build/src/build/matches/test.rs]: "exhaustive" is also a technical term and the right one to use here, and "inexhaustive" is its correct negation.

**Comment by bjorn3** [comment]
I'm done reviewing. Apart from the remaining open review comments I think it there are no other mistakes here.

**Comment by RalfJung** [comment]
Given the amount of false positives here, we should make more explicitly ask for not just submitting tool-based spellcheck PRs? The reviewing load is not in good relation to the benefit. I recall @oli-obk and others having that discussion about punctuation PRs, not sure if any official policy was ever written based on that?

**Comment by oli-obk** [comment]
We do have this policy. I was hoping the PRs were just one offs, but it is indeed getting too much again.

As per https://github.com/rust-lang/rust/issues/58619 I am closing this PR and will close all upcoming ones, too.

If you do want to continue working on this, please consider developing tooling that can be applied without false positives like our formatting tooling.

**Comment by nyurik** [comment]
Thanks @compiler-errors and @bjorn3 for the review!  I will resolve all the simple issues you raised.

@RalfJung that's a fair point.  I did add almost 2,000 words to my dictionary as valid, and those that slipped became false positives. That said -- I verified each one of them by hand, and I was fairly certain that the proposed changes are OK. My apologies for making mistakes.

**Comment by nyurik** [comment]
@oli-obk would storing a list of words as part of this (or other) repo work?  False positives only exist because of the natural dictionary limitations, so storing well known per-project/per-domain lists should fix it.

Per #58619 you posted -- `Oh, and I haven't mentioned it, but it can't possibly annoy anyone – typo/spelling/grammatical fixes should be fine too.` (per @alexreg) -- isn't that what i was following here?

One more thing -- would it make sense to not merge it after all the reviews were done? If there is a policy against spellchecking, I will be happy to not submit any more such PRs, but it seems this one is fairly innocuous and has already been vetted by two contributors.

**Comment by RalfJung** [comment]
No worries, mistakes happen. :) My concern is more generally with the cost-benefit trade-off for PRs like this. They touch a lot of code and hence automatically ping a lot of people and take sparse reviewer time away. And yet they only temporarily actually reduce the number of typos in the codebase. I'm not sure having a constant stream of such PRs (fixing the constant stream of new typos being added) is worth the constant load this puts on reviewers (reviewer time is a very sparse resource).

OTOH, being able to fix a particular spelling mistake *once and for all* (so `./x.py fmt` takes care of it in the future) would be a very different situation. That's why @oli-obk and me keep asking for tools.

Note that `alexreg` was the person doing the style PR changes back then, so this represents their view, not the view of the team(s).

**Comment by RalfJung** [comment]
FWIW though I also agree that if @bjorn3 already spent the reviewer time and agrees the remaining changes are clearly fixing typos and there are no ambiguities, then there's no point in throwing away that work. 160 line diff isn't that bad re: risk of conflicts.

**Comment by oli-obk** [comment]
Yea, the work here has been done, so I'd be ok with merging it, but that's also what happend to the previous 5 PRs or so in the last two weeks.

I'm not going to block you or Bjorn3 moving ahead with this PR, but I'm going to block all future PRs until we get tooling.

**Comment by oli-obk** [comment]
> would storing a list of words as part of this (or other) repo work? False positives only exist because of the natural dictionary limitations, so storing well known per-project/per-domain lists should fix it.

Yes, totally. As long as we can automatically apply it with `./x.py fmt` and at some point gate CI on it, I'm for it.

The tooling must be easy to use on all platforms though, so there may be limitations to what tools can be used. Requiring contributors to install something is likely not going to get accepted. Python or Rust programs on the other hand will cause little friction

**Comment by compiler-errors** [comment]
Given @bjorn3's and my reviews, and the comments above by @oli-obk and @RalfJung who are not opposed to this one landing, I'll reopen this.

I would also be quite elated to see a style linter in `./x.py fmt`. I will also keep #58619 in mind if I see future style/formatting PRs in the review queue.

**Comment by compiler-errors** [comment]
since bjorn3 said they were done reviewing too, and the only nits I see are minimal:
@bors r+

File: compiler/rustc_apfloat/src/ieee.rs (hunk 5 above)
Comment (RalfJung) [reply compiler/rustc_apfloat/src/ieee.rs]: FWIW this PR still changes apfloat

Comment (compiler-errors) [reply compiler/rustc_apfloat/src/ieee.rs]: @ralfjung: this doesn't seem to change apfloat on the most up to date set of changes?

Comment (RalfJung) [reply compiler/rustc_apfloat/src/ieee.rs]: Oh weird, not sure what happened... I did load the page after you r+'d but it still showed an old diff I guess? Must be that 'eventual consistency' they always talk about...

Well sorry for the noise then.

**Comment by nyurik** [comment]
@RalfJung I just found out about [cargo-spellcheck](https://github.com/drahnr/cargo-spellcheck) - might be a good solution long term

**Pull request merged**